### PR TITLE
WIP Fix/header button translation [NP-1884]

### PR DESCRIPTION
**Pull request by davidsauntson**
Fixes a bug in the mobile header where the title of the search expand / collapse button was not translated in the javascript by:
 - creating translations for open and close states
 - storing these in `data-` attributes on the button
 - updating the js to use these attributes when updating the button title

Waiting for the Welsh translation of 'close search' but ready for review.

### CODE REVIEW

File: haml/_header.html.haml [hunk 1]
```diff
@@ -3,6 +3,12 @@
   sign_in_out ||= {}
   main_content_anchor_link ||= "#cads-main-content"
   homepage_url ||= "/"
+  button_attr = {
+    class: %w[cads-header__search-reveal js-cads-search-reveal cads-icon_search],
+    title: t("cads.header.open_search"),
```
Comment (luke-hill) [inline haml/_header.html.haml:8]: is this value needed now? Given your JS switches it below?

File: testing/features/header.feature [hunk 2]
```diff
@@ -60,12 +60,12 @@ Feature: Header component
...
       Given a Standard Header component is on the page
 
-    @small_screen @failing @NP-1480
+    @small_screen @failing @NP-1885
     Scenario: English Header has be expanded to show the full search bar
       Then I am able to expand the search bar
       And I am able to collapse the search bar
 
-    @small_screen @failing @NP-1480
+    @small_screen @failing @NP-1885
```
Comment (luke-hill) [inline testing/features/header.feature:68]: we can change this tag to `@small_screen @future_release @v4.1.0+`

**Review by luke-hill — APPROVED** [review]
Seems legit. Couple of minor points but non blocking / could be wrong.

File: haml/_header.html.haml (hunk 1 above)
Comment (davidsauntson) [reply haml/_header.html.haml:8]: It is needed - for the initial render

File: testing/features/header.feature (hunk 2 above)
Comment (davidsauntson) [reply testing/features/header.feature:68]: It's going to be in the next release though so I'll change it to `4.0.4`

**Review by davidrapson — APPROVED** [review]
👍 But you've got a lint error from `i18n-tasks`, asking you to run `normalize`.

**Pull request merged**